docs/contracts/transient-storage.rst

```diff
@@ -0,0 +1,166 @@
+The caller of an external call may bypass that mechanism using a ``try ... catch`` block.
+This mimics the behavior of persistent storage.
+
+.. note::
+    Currently, the compiler can parse ``transient`` as a data location, however it is not
+    defined as a keyword of the language yet. This means that the use of ``transient``
+    is backwards-compatible and does not break previous code that eventually used it as an identifier.
+
+*********************************************************************
+Composability of Smart Contracts and the Caveats of Transient Storage
+*********************************************************************
+
+Given the caveats mentioned in the specification of EIP-1153,
+in order to preserve the composability of your smart contract,
+utmost care is recommended for more advanced use cases of transient storage.
+
+For smart contracts, composability is a very important design to achieve a self-contained behaviour,
+such that multiple calls into individual smart contracts can be composed to more complex applications.
+So far the EVM largely guaranteed composable behaviour, since multiple calls into a smart contract
+within a complex transaction are virtually indistinguishable from multiple calls to the contract
+stretched over several transactions. However, transient storage allows a violation to this principle
+and incorrect use may lead to complex bugs that only surface when used across several calls.
+
+Let's illustrate the problem with a simple example:
+
+.. code-block:: solidity
+
+    // SPDX-License-Identifier: GPL-3.0
+    pragma solidity ^0.8.27;
+
+    contract MulService {
+        uint transient multiplier;
+        function setMultiplier(uint mul) external {
+            multiplier = mul;
+        }
+
+        function multiply(uint value) external view returns (uint) {
+            return value * multiplier;
+        }
+    }
+
+If the example used memory or storage to store the multiplier, it would be fully composable.
+It would not matter whether you split the sequence into separate transactions or grouped them in some way.
+You would always get the same result. This enables use cases such as batching calls from multiple transactions
+together to reduce gas costs. Transient storage potentially breaks such use cases since composability can no longer be taken for granted.
+
+Note however, that the lack of composability is not an inherent property of transient storage.
+It could have been preserved if the rules for resetting its content were slightly adjusted.
+Currently the clearing happens for all contracts at the same time, when the transaction ends.
+If instead it was cleared for a contract as soon as no function belonging to it remained active
+on the call stack (which could mean multiple resets per transaction), the issue would disappear.
+In the example above it would mean clearing transient storage after each of the calls.
+
```

Comment:
I'd drop this - little reason to dwell on it, it works how it's specified now - our docs are not the place to argue that things could have been different.